examples/40_paper/2018_kdd_rijn_example.py

```diff
@@ -8,9 +8,116 @@
+# addressed by defining a ConfigSpace. For a more elaborated example that uses
+# this, please see:
+# https://github.com/janvanrijn/openml-pimp/blob/d0a14f3eb480f2a90008889f00041bdccc7b9265/examples/plot/plot_fanova_aggregates.py # noqa F401
+
+suite = openml.study.get_suite('OpenML100')
+flow_id = 7707
+parameter_filters = {
+    'sklearn.svm.classes.SVC(17)_kernel': 'sigmoid'
+}
+evaluation_measure = 'predictive_accuracy'
+limit_per_task = 500
+n_trees = 16
+
+fanova_results = []
+# we will obtain all results from OpenML per task. Practice has shown that this places the bottleneck on the
+# communication with OpenML, and for iterated experimenting it is better to cache the results in a local file.
+for idx, task_id in enumerate(suite.tasks):
+    logging.info('Starting with task %d (%d/%d)' % (task_id, idx+1, len(suite.tasks)))
+    evals = openml.evaluations.list_evaluations_setups(
+        evaluation_measure, flow=[flow_id], task=[task_id], size=limit_per_task, output_format='dataframe')
+
+    performance_column = 'value'
+    # make a DataFrame consisting of all hyperparameters (which is a dict in setup['parameters']) and the performance
+    # value (in setup['value']). The following line looks a bit complicated, but combines 2 tasks: a) combine
+    # hyperparameters and performance data in a single dict, b) cast hyperparameter values to the appropriate format
+    # Note that
+    try:
+        setups_evals = pd.DataFrame([dict(**{name: json.loads(value) for name, value in setup['parameters'].items()},
+                                          **{performance_column: setup[performance_column]})
+                                     for _, setup in evals.iterrows()])
+    except json.decoder.JSONDecodeError as e:
+        logging.warning('Task %d error: %s' % (task_id, e))
+        continue
+    # apply our filters, to have only the setups that comply to the hyperparameters we want
+    for filter_key, filter_value in parameter_filters.items():
+        setups_evals = setups_evals[setups_evals[filter_key] == filter_value]
+    # in this simplified example, we only display numerical and float hyperparameters. For categorical hyperparameters,
+    # the fanova library needs to be informed by using a configspace object.
+    setups_evals = setups_evals.select_dtypes(include=['int64', 'float64'])
+    # drop rows with unique values. These are by definition not an interesting hyperparameter, e.g., `axis`, `verbose`
+    setups_evals = setups_evals[[c for c in list(setups_evals)
+                                 if len(setups_evals[c].unique()) > 1 or c == performance_column]]
+    # determine x values to pass to fanova library
+    parameter_names = [pname for pname in setups_evals.columns.values if pname != performance_column]
+    evaluator = fanova.fanova.fANOVA(
+        X=setups_evals[parameter_names].values, Y=setups_evals[performance_column].values, n_trees=n_trees)
```

Comment:
Could you please use https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_numpy.html instead of `dataframe.values`?